src/game/common/rts/specialpower.cpp

```diff
@@ -62,6 +62,98 @@ bool SpecialPowerStore::Can_Use_Special_Power(Object *obj, const SpecialPowerTem
     return true;
 }
 
+SpecialPowerTemplate *SpecialPowerStore::Find_Special_Power_Template_Private(Utf8String name)
+{
+    for (unsigned int i = 0; i < m_specialPowerTemplates.size(); i++) {
+        if (m_specialPowerTemplates[i]->Get_Name() == name) {
+            return m_specialPowerTemplates[i];
+        }
+    }
+
+    return nullptr;
+}
+
+const SpecialPowerTemplate *SpecialPowerStore::Find_Special_Power_Template(Utf8String name)
+{
+    return Find_Special_Power_Template_Private(name);
+}
+
+const SpecialPowerTemplate *SpecialPowerStore::Get_Special_Power_Template_By_Index(unsigned int index)
+{
+    if (index < m_specialPowerTemplates.size()) {
+        return m_specialPowerTemplates[index];
+    } else {
+        return nullptr;
+    }
+}
+
+int SpecialPowerStore::Get_Num_Special_Powers()
+{
+    return m_specialPowerTemplates.size();
+}
+
+void SpecialPowerStore::Parse_Special_Power_Definition(INI *ini)
+{
+    Utf8String str = ini->Get_Next_Token();
+    SpecialPowerTemplate *power = g_theSpecialPowerStore->Find_Special_Power_Template_Private(str);
+
+    if (ini->Get_Load_Type() == INI_LOAD_CREATE_OVERRIDES) {
+        if (power != nullptr) {
+            SpecialPowerTemplate *final_override = static_cast<SpecialPowerTemplate *>(power->Friend_Get_Final_Override());
+            power = new SpecialPowerTemplate();
+            *power = *final_override;
+            final_override->Set_Next(power);
+            power->Set_Is_Allocated();
+        } else {
+            power = new SpecialPowerTemplate();
+            const SpecialPowerTemplate *default_power =
+                g_theSpecialPowerStore->Find_Special_Power_Template("DefaultSpecialPower");
+
+            if (default_power != nullptr) {
+                *power = *default_power;
+            }
+
+            power->Set_Name_ID(str, ++g_theSpecialPowerStore->m_nextSpecialPowerID);
+            power->Set_Is_Allocated();
+            g_theSpecialPowerStore->m_specialPowerTemplates.push_back(power);
+        }
+    } else {
+        if (power != nullptr) {
+            throw CODE_06;
+        }
+
+        power = new SpecialPowerTemplate();
+        const SpecialPowerTemplate *default_power =
+            g_theSpecialPowerStore->Find_Special_Power_Template("DefaultSpecialPower");
+
+        if (default_power != nullptr) {
+            *power = *default_power;
+        }
+
+        power->Set_Name_ID(str, ++g_theSpecialPowerStore->m_nextSpecialPowerID);
```

Comment:
`power->Set_Is_Allocated();` missing?

Reply:
No, it specifically doesn't exist (I had it there previously and it caused things to break until I realized my mistake).